_data/data.js

```diff
@@ -280,6 +382,14 @@ const resizeImage = async function (filename) {
     Path.relative(IMAGE_DIRECTORY, directory),
   );
   const images = {};
+
+  for (const [newFile, oldFile] of sameImageMap) {
+    if (filename === newFile) {
```

Comment:
This should work with the current code, but it's based on the assumption that the image that's downloaded earlier would be resized earlier. Let's say file A, B, C are all pointed to the same URL. A is downloaded first, so you will build a map like this:

```
[
  [A, B],
  [A, C],
]
```

If B is the first to be resized, it's fine by itself. Then A won't reuse B's resize output. You may use bi-directional relationship to fix this, so the map is more like this:

```
[
  [A, B],
  [B, A],
  [A, C],
  [C, A],
]
```

However, C still won't reuse B's output because they are not connected. If you want to connect B and C, you will need a map like this:

```
[
  [A, B],
  [B, A],
  [A, C],
  [C, A],
  [B, C],
  [C, B],
]
```

You can definitely construct such map, but is there a better way to deal with this?